<!-- -->
<launch>


<include file="$(find ahmedrrt)/launch/real_gmapping.launch"/>

  <!-- Publishes the voxel grid to rviz for display -->
  <node pkg="costmap_2d" type="costmap_2d_markers" name="voxel_visualizer">
    <remap from="voxel_grid" to="costmap/voxel_grid"/>
  </node>

  <!-- Run the costmap node -->
  <node name="costmap_node" pkg="costmap_2d" type="costmap_2d_node" >
    <rosparam file="$(find ahmedrrt)/param/cost_params_real.yaml" command="load" ns="costmap" />
    <param name="costmap/static_layer/track_unknown_space" value="true"/>
    <param name="costmap/obstacle_layer/track_unknown_space" value="true"/> 
    <param name="costmap/publish_frequency" value="2.0"/> 
    <param name="costmap/always_send_full_costmap" value="true"/> 
  </node>

<node pkg="ahmedrrt" type="rrt_server.py" name="rrt_server">
<remap from="map" to="costmap_node/costmap/costmap"/>
<param name="eta" value="0.2"/>
<param name="goal_tolerance" value=".5"/>
<param name="smoothing_max_iter" value="100"/>
</node>

<node pkg="ahmedrrt" type="move_base.py" name="move_base">
<remap from="/mobile_base/commands/velocity" to="/cmd_vel_mux/input/navi"/>
</node>
  
  
<node pkg="rviz" type="rviz" name="rviz" args="-d $(find ahmedrrt)/launch/include/single.rviz">
<remap from="move_base_simple/goal" to="move_base_simple/goal"/>
</node>

</launch>
